### PR TITLE
Expand powdered snow advancement to check modded armor

### PR DESCRIPTION
**Pull request by pupnewfster**
Override vanilla's walk on powdered snow advancement to check against `IItemExtension#canWalkOnPowderedSnow` in order to support triggering the advancement for modded items that use that hook.

### CODE REVIEW

File: src/main/java/net/neoforged/neoforge/common/data/internal/NeoForgeAdvancementProvider.java [hunk 1]
```diff
@@ -122,17 +126,39 @@ private static boolean predicateMatches(ItemPredicate predicate, ItemLike... tar
...
+                        EntityPredicate replacement = new EntityPredicate(
+                                entityPredicate.entityType(),
+                                entityPredicate.distanceToPlayer(),
+                                entityPredicate.movement(),
+                                entityPredicate.location(),
+                                entityPredicate.effects(),
+                                entityPredicate.nbt(),
+                                entityPredicate.flags(),
+                                Optional.empty(),
+                                Optional.of(subPredicate),
```
Comment (Technici4n) [inline src/main/java/net/neoforged/neoforge/common/data/internal/NeoForgeAdvancementProvider.java:153]: Should we assert that there was no `subPredicate` before?

Comment (pupnewfster) [reply src/main/java/net/neoforged/neoforge/common/data/internal/NeoForgeAdvancementProvider.java:153]: You mean just to make it easier to notice if it breaks when porting mc versions?

Comment (Technici4n) [reply src/main/java/net/neoforged/neoforge/common/data/internal/NeoForgeAdvancementProvider.java:153]: Yeah exactly.

Comment (pupnewfster) [reply src/main/java/net/neoforged/neoforge/common/data/internal/NeoForgeAdvancementProvider.java:153]: Done

**Review by Technici4n — APPROVED** [review]

**Pull request merged**